src/core/MOM_forcing_type.F90

```diff
@@ -3019,6 +3030,24 @@ subroutine allocate_mech_forcing_by_group(G, forces, stress, ustar, shelf, &
   !These fields should only on allocated when iceberg area is being passed through the coupler.
   call myAlloc(forces%area_berg,isd,ied,jsd,jed, iceberg)
   call myAlloc(forces%mass_berg,isd,ied,jsd,jed, iceberg)
+
+  !These fields should only be allocated when waves
+  call myAlloc(forces%ustk0,isd,ied,jsd,jed, waves)
+  call myAlloc(forces%vstk0,isd,ied,jsd,jed, waves)
+  if (present(waves)) then; if (waves) then; if (.not.associated(forces%ustkb)) then       
+    if (.not.(present(num_stk_bands))) call MOM_error(FATAL,"Requested to &
+      initialize with waves, but no waves are present.") 
+    allocate(forces%stk_wavenumbers(num_stk_bands)) 
+    forces%stk_wavenumbers(:) = 0.0    
+    allocate(forces%ustkb(isd:ied,jsd:jed,num_stk_bands)) 
+    forces%ustkb(isd:ied,jsd:jed,:) = 0.0
+    endif; endif; endif
```

Comment:
The indenting of the endif statement on line 3044 should match its counterpart if statement on line 3037. Also, as a minor point of style, elsewhere in the MOM6 code we use a space before and after the semicolons, to help distinguish them in greps from any semicolons used in comments in standard written English, so "endif; endif; endif" on line 3044 should be "endif ; endif ; endif".